In [ ]:
from google.colab import drive
drive.mount('/gdrive')
%cd /gdrive

Mounted at /gdrive
/gdrive


In [ ]:
#Reading original data
import pandas as pd
data=pd.read_csv('/gdrive/My Drive/IMDB Dataset.csv')

In [ ]:
#Reading pre processed data
pre_data=pd.read_csv('/gdrive/My Drive/Movie Reviews sentiment analysis/final_processed_data.csv')
pre_data.head()

Unnamed: 0                                                  0
0           0  one reviewers mentioned watching oz episode ho...
1           1  wonderful little production filming technique ...
2           2  thought wonderful way spend time hot summer we...
3           3  basically family little boy jake thinks zombie...
4           4  petter mattei love time money visually stunnin...

Since sentiment values are categorical values so we convert it into numerical values using replace method.

In [ ]:
#changing positive to 1 and negative to 0
data['sentiment'].replace('positive',1,inplace=True)
data['sentiment'].replace('negative',0,inplace=True)


After replacing , we are concatinating pre_data review column and data sentiment column to get a complete pre-processed data for model development.

In [ ]:
#Concatinating the columns
pre_data=pd.concat([pre_data['0'],data['sentiment']],axis=1)
pre_data.head(10)

0  sentiment
0  one reviewers mentioned watching oz episode ho...          1
1  wonderful little production filming technique ...          1
2  thought wonderful way spend time hot summer we...          1
3  basically family little boy jake thinks zombie...          0
4  petter mattei love time money visually stunnin...          1
5  probably time favorite movie story selflessnes...          1
6  sure would like see resurrection dated seahunt...          1
7  show amazing fresh innovative idea first aired...          0
8  encouraged positive comments film looking forw...          0
9  like original gut wrenching laughter like movi...          1

In [ ]:
#Changing column names for easy convenience
pre_data.columns=['Reviews','Sentiment']
pre_data.head(10)

Reviews  Sentiment
0  one reviewers mentioned watching oz episode ho...          1
1  wonderful little production filming technique ...          1
2  thought wonderful way spend time hot summer we...          1
3  basically family little boy jake thinks zombie...          0
4  petter mattei love time money visually stunnin...          1
5  probably time favorite movie story selflessnes...          1
6  sure would like see resurrection dated seahunt...          1
7  show amazing fresh innovative idea first aired...          0
8  encouraged positive comments film looking forw...          0
9  like original gut wrenching laughter like movi...          1

Splitting the data into train dataset and test dataset 

In [ ]:
X=pre_data['Reviews']
Y=pre_data['Sentiment']

#train and test dataset
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(X,Y,test_size=0.2,random_state=2)

In [ ]:
#Printing the shapes of train and test dataset
print(x_train.shape)
print(y_train.shape)
print(x_test.shape)
print(y_test.shape)


(40000,)
(40000,)
(10000,)
(10000,)


## **TF-IDF**

1. TF-IDF which stands for Term Frequency – Inverse Document Frequency.  
2. It is one of the most important techniques used for information retrieval to represent how important a specific word or phrase is to a given document.
3. Implement using TfidfVectorizer from sklearn library.




In [ ]:
#converting the words into vectors using TfIdfvectorizer
from sklearn.feature_extraction.text import TfidfVectorizer
tf_idf=TfidfVectorizer()
x_train_tf=tf_idf.fit_transform(x_train)

In [ ]:
#2.RandomForestclassifier
from sklearn.ensemble import RandomForestClassifier
rand_forest=RandomForestClassifier(n_estimators=300,criterion='gini')
rand_forest.fit(x_train_tf,y_train)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=300,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [ ]:
#prediction for randomforest
x_test_tf=tf_idf.transform(x_test).toarray()
pred_forest=rand_forest.predict(x_test_tf)
print(pred_forest)

[0 0 1 ... 0 1 0]


In [ ]:
#Accuracy metrics
from sklearn.metrics import classification_report,confusion_matrix,accuracy_score

# 1. Confusion matrix
confu_matrix=confusion_matrix(y_test,pred_forest)
print('Confusion for the model is :\n ',confu_matrix)

# 2. Accuracy score
accu_score=accuracy_score(y_test,pred_forest)
print('The accuracy score for the model is : ',accu_score)
# 3. Classification report
report=classification_report(y_test,pred_forest)
print('The classification report for the model :\n',report)

Confusion for the model is :
  [[4323  656]
 [ 655 4366]]
The accuracy score for the model is :  0.8689
The classification report for the model :
               precision    recall  f1-score   support

           0       0.87      0.87      0.87      4979
           1       0.87      0.87      0.87      5021

    accuracy                           0.87     10000
   macro avg       0.87      0.87      0.87     10000
weighted avg       0.87      0.87      0.87     10000



Comparing Bag of words and TF_IDF, TF-IDF gives a better accuracy of 0.87 in Random Forest Classifier model.